conda/conda
# fix update idempotency; add tests for channel priority

There are 2 test case for channel priority

1. update --all with channel priority, pass
2. update one package with channel priority (previously failed)

Will fix issue #3303 

The failure comes from [here](https://github.com/conda/conda/blob/master/conda/cli/install.py#L244). In this situation, it just check version and build number. However it should check channel as well, otherwise, update --all and update single_package may not be consistent

@kalefranz @mcg1969
